code/game/objects/effects/contraband.dm

```diff
@@ -78,6 +78,9 @@
 
 	var/obj/structure/sign/poster/selected = pick(approved_types)
 
+	if(SSevents.holidays && prob(30))  // more festive posters on festive days!
+		selected = /obj/structure/sign/poster/official/festive
```

Comment:
this should only apply if the holiday has a poster

Reply:
it has a generic poster varient so it doesnt mess with the random poster code too much ^^